src/compile/layer.ts

```diff
@@ -184,12 +189,13 @@ export class LayerModel extends Model {
     return applyConfig({}, cellConfig, FILL_STROKE_CONFIG.concat(['clip']));
   }
 
+  // TODO: Support same named selections across children.
```

Comment:
Can you add the name of the model to the key to make this work? Call `model.getName('selection name')`.